ros/angel_system_nodes/angel_system_nodes/task_monitor.py

```diff
@@ -282,6 +291,34 @@ def listener_callback(self, activity_msg):
             # No timer associated with this state
             pass
 
+    def query_task_graph_callback(self, request, response):
+        """
+        Populate the `QueryTaskGraph` response with the task list
+        and return it.
+        """
+        log = self.get_logger()
+        task_g = TaskGraph()
+
+        task_g.task_nodes = []
+        for t in self._task.steps:
+            t_node = TaskNode()
+
+            t_node.uid =  str(uuid.uuid4())
```

Comment:
This is going to create new UUIDs for the same tasks every time this is called. Either UUIDs should be assigned on construction, or be created "lazily" when this service is called for the first time (the former is probably simpler). UUIDs ideally might be consistent across restarts if they are based in some way on task data, e.g. the md5 checksum of the task name.

Reply:
Actually, maybe one step better: since this response is static (the task graph is currently not changing during runtime), this message could be created once at construction and simply returned when this callback in invoked.

Reply:
Re: md5 checksum, is it possible that we would have repeated tasks? Like step 5 and 10 are the same? And would we want those to have the same uid?

Reply:
> Actually, maybe one step better: since this response is static (the task graph is currently not changing during runtime), this message could be created once at construction and simply returned when this callback is invoked.

I figured the use for this would be called once on startup (which is the other task I have to do)... if I'm following you correctly?

Reply:
> Re: md5 checksum, is it possible that we would have repeated tasks? Like step 5 and 10 are the same? And would we want those to have the same uid?

Disregard me then, I don't think we need to have consistent UUIDs across processes for now so it's not really worth the effort to iron out those details with such a mechanism right now.

---

> I figured the use for this would be called once on startup (which is the other task I have to do)... if I'm following you correctly?

I presume you're also thinking about the HoloLens2 app's invocation of this which may ideally only call this "once". My comment here was just along the lines that this function from the source of truth is not idempotent. Logically it should return the same thing for the duration of the process but it currently does not.